File: src/java_answer/Lessons12924.java
package java_answer;

import static java.lang.System.out;

/**
 * @author dave.th
 * https://programmers.co.kr/learn/courses/30/lessons/12924
 */
public class Lessons12924 {
    public static void main(String[] args) {
        out.println((new Lessons12924()).solution(15));
    }

    private int solution(int n) {
       int count = 0;
       for(int i=1; i<=n; i++){
           int sum = 0;
           for(int j = i; j<=n && sum < n; j++){
               sum += j;
               if(sum == n) count++;
           }
       }

       return count;
    }
}

File: src/scala_answer/Lessons12924.scala
package scala_answer

/**
  * @author dave.th
  * https://programmers.co.kr/learn/courses/30/lessons/12924
  */
object Lessons12924 {
  def main(args: Array[String]) : Unit = {
    println(solution(15))
  }

  def solution(n : Int) : Int = {
    var count = 0
    for(i <- 1 to n) {
      var sum = 0
      for(j <- i to n; if sum < n) {
        sum = sum + j
        if(sum == n) count = count + 1
      }
    }

    count
  }
}
